src/components/profile.ts

```diff
@@ -0,0 +1,153 @@
+import { openDB } from './db';
+
+// since interfaces are for compile time there is no easy way to extract keys
+// so this userProfileKeys is needed
+export const userProfileKeys = [
+  'about_me',
+  'birth_date',
+  'preferred_name',
+  'preferred_pronouns',
+  'term',
+  'year',
+  'faculty',
+  'program',
+  'specialization'
+];
+
+export interface UserProfile {
+  about_me?: string;
+  birth_date?: string;
+  preferred_name?: string;
+  preferred_pronouns?: string;
+  term?: string;
+  year?: number;
+  faculty?: string;
+  program?: string;
+  specialization?: string;
+}
+
+const validFaculties = ['Mathematics', 'Engineering', 'Arts', 'Health', 'Science'];
+const validTerms: string[] = ['1A', '1B', '2A', '2B', '3A', '3B', '4A', '4B', 'MASc', 'PHD'];
+const validYears: number[] = [];
+const yearStart = 2000;
+const yearEnd = 2030;
```

Comment:
this is a finicky upper bound.

although 2030 does seem far off, I'm graduating in 2025, which means the year after that there's going to be people with graduation dates 2031.